web/src/app/util/SpeedDial.js

```diff
@@ -1,74 +1,70 @@
-import React from 'react'
+import React, { useState } from 'react'
 import p from 'prop-types'
-import SpeedDialAction from '@material-ui/lab/SpeedDialAction'
-import SpeedDialIcon from '@material-ui/lab/SpeedDialIcon'
-import SpeedDial from '@material-ui/lab/SpeedDial'
+import { SpeedDial, SpeedDialIcon, SpeedDialAction } from '@material-ui/lab'
+import { makeStyles } from '@material-ui/styles'
 
-export default class ScheduleNewOverrideFAB extends React.PureComponent {
-  static propsTypes = {
-    label: p.string.isRequired,
-    actions: p.arrayOf(
-      p.shape({
-        icon: p.element.isRequired,
-        onClick: p.func.isRequired,
-        label: p.string.isRequired,
-      }),
-    ).isRequired,
-  }
+const useStyles = makeStyles(theme => ({
+  speedDial: {
+    position: 'fixed',
+    bottom: '2em',
+    right: '2em',
+    zIndex: 9001,
+  },
+  staticTooltipLabel: {
+    whiteSpace: 'nowrap',
+  },
+}))
 
-  state = {
-    open: false,
-  }
+export default function CustomSpeedDial(props) {
+  const [open, setOpen] = useState(false)
+  const classes = useStyles()
 
-  shownState = false
-  _shownTimeout = -1
+  const doToggle = () => setOpen(!open)
```

Comment:
let's use the `handle` naming convention set by `jsx-handler-names`, where we name `onEvent` funcs to `handleNounVerb`, i.e. `handleDialOpen` or similar